gramps/gen/lib/grampstype.py

```diff
@@ -151,6 +151,16 @@ def set_object_state(self, attr_dict):
         else:
             self.__string = ""
 
+    def __set_dict(self, value):
```

Comment:
Why doesn't `__set_dict` call `__set_tuple`?

The method would then only need to consist of a single line:
```
self.__set_tuple((value.value, value.string))
```